conans/client/build/cmake_flags.py

```diff
@@ -125,6 +125,68 @@ def build_type_definition(build_type, generator):
     return {}
 
 
+class CMakeDefinitions(object):
+    """
+    Create list of unrepeated definitions and discard the ones without value as result
+    """
+
+    def __init__(self, definitions=None):
+        if definitions:
+            if isinstance(definitions, list):
+                for item in definitions:
+                    assert isinstance(item, tuple), self._init_assert_msg()
+                definitions = OrderedDict(definitions)
+            assert isinstance(definitions, OrderedDict), self._init_assert_msg()
+        self._definitions = definitions or OrderedDict()
+
+    def _init_assert_msg(self):
+        return "Definitions argument needs to be list of tuples or an OrderedDict to preserve the " \
+               "order of items"
+
+    def set(self, key, value):
+        """
+        Set a definition with a valid value checking that the key already exists (it is an
+        expected key), and the value is different form the previous one
+        :param key: Definition name
+        :param value: Definition value
+        """
+        self._set(key, value)
+
+    def _set(self, key, value, new=False):
+        if not new:
+            assert key in self._definitions, "Key '%s' not previously set in dictionary" % key
+            assert self._definitions[key] is None, \
+                "Key '%s' already has a value assigned: '%s'" % (key, self._definitions[key])
+        else:
+            assert key not in self._definitions, "Key '%s' previously set in dictionary" % key
+        self._definitions[key] = value
+
+    def update(self, definitions):
+        """
+        Update the definitions creating new keys and checkign they weren't previously added
+        :param definitions: Dictionary with definition names as keys and their values
+        """
+        for key, value in definitions.items():
+            self._set(key, value, True)
+
+    def get(self, key):
```

Comment:
This is anti-pythonic, if it is the case, implement ``__getitem__`` and ``__setitem__``